lib/active_merchant/billing/gateways/nmi.rb

```diff
@@ -248,6 +250,18 @@ def add_merchant_defined_fields(post, options)
         end
       end
 
+      def add_three_d_secure(post, options)
+        return unless options[:three_d_secure]
+
+        if (three_d_secure = options[:three_d_secure])
+          post[:eci] = three_d_secure[:eci]
+          post[:cavv] = three_d_secure[:cavv]
+          post[:xid] = three_d_secure[:xid]
+          post[:three_ds_version] = three_d_secure[:version]
+          post[:source_transaction_id] = three_d_secure[:ds_transaction_id]
```

Comment:
Should this map to `directory_server_id`, instead of `source_transaction_id`?

Reply:
Yes! Good catch! This has been updated.